flink-python/src/main/java/org/apache/flink/python/client/PythonShellParser.java

```diff
@@ -0,0 +1,322 @@
+		.build();
+
+	private static final Option OPTION_TM_MEMORY = Option
+		.builder("tm")
+		.required(false)
+		.longOpt("taskManagerMemory")
+		.hasArg()
+		.desc("Memory per TaskManager Container with optional unit (default: MB)")
+		.build();
+
+	// cluster types
+	private static final String LOCAL_RUN = "local";
+	private static final String REMOTE_RUN = "remote";
+	private static final String YARN_RUN = "yarn";
+
+	// Options that will be used in mini cluster.
+	private static final Options LOCAL_OPTIONS = getLocalOptions(new Options());
+
+	// Options that will be used in remote cluster.
+	private static final Options REMOTE_OPTIONS = getRemoteOptions(new Options());
+
+	// Options that will be used in yarn cluster.
+	private static final Options YARN_OPTIONS = getYarnOptions(new Options());
+
+	public static void main(String[] args) {
+		if (args.length < 1) {
+			printError("You should specify cluster type or -h | --help option");
+			System.exit(1);
+		}
+		String command = args[0];
+		List<String> commandOptions = null;
+		try {
+			switch (command) {
+				case LOCAL_RUN:
+					commandOptions = parseLocal(args);
+					break;
+				case REMOTE_RUN:
+					commandOptions = parseRemote(args);
+					break;
+				case YARN_RUN:
+					commandOptions = parseYarn(args);
+					break;
+				case "-h":
+				case "--help":
+					printHelp();
+					break;
+				default:
+					printError(String.format("\"%s\" is not a valid cluster type or -h | --help option.\n", command));
+					System.exit(1);
+			}
+			if (commandOptions != null) {
+				for (String option : commandOptions) {
+					System.out.print(option);
+					System.out.print('\0');
+				}
+			}
+		} catch (Throwable e) {
+			printError("Error while running the command.");
+			e.printStackTrace();
+			System.exit(1);
+		}
+	}
+
+	private static void buildGeneralOptions(Options options) {
+		options.addOption(OPTION_HELP);
+	}
+
+	private static Options getLocalOptions(Options options) {
+		buildGeneralOptions(options);
+		return options;
+	}
+
+	private static Options getRemoteOptions(Options options) {
+		buildGeneralOptions(options);
+		return options;
+	}
+
+	private static Options getYarnOptions(Options options) {
+		buildGeneralOptions(options);
+		options.addOption(OPTION_CONTAINER);
+		options.addOption(OPTION_JM_MEMORY);
+		options.addOption(OPTION_NAME);
+		options.addOption(OPTION_QUEUE);
+		options.addOption(OPTION_SLOTS);
+		options.addOption(OPTION_TM_MEMORY);
+		return options;
+	}
+
+	/**
+	 * Prints the error message and help for the client.
+	 *
+	 * @param msg error message
+	 */
+	private static void printError(String msg) {
+		System.err.println(msg);
+		System.err.println("Valid cluster type are \"local\", \"remote <hostname> <portnumber>\", \"yarn\".");
+		System.err.println();
+		System.err.println("Specify the help option (-h or --help) to get help on the command.");
+	}
+
+	/**
+	 * Prints the help for the client.
+	 */
+	private static void printHelp() {
```

Comment:
I think we should have deffrence help message for `local`, `remote` and `yarn`, otherwise we do not need add the `-h` option for `local`, `remote` and `yarn`. What do you think?